Cell fired upon aj

AntonioJacksonII opened the pull request:
Added fired_upon? & fire_upon methods with associated tests.

JBelta commented:
lib/cell rb: 14 and 18. - I keep forgetting how "==" isn't the same as "=". I should have done push.

cell_test rb: 31. - I typed in the brackets around cruiser because I thought the value would return as an array. In the test I ran before hand it always returned as a Fail because the return was in brackets. Did this test pass?

lib/ship.rb rb:4. - Just making sure I get this right, but we have health as an attr_accessor because we want it's value to change, but weren't we suppose to avoid attr_accessor? I thought Megan told us not to.

AntonioJacksonII commented:
lib/cell rb: 14 and 18 - The issue here wasn't the `=`, `==`, or `.push`,  it was the fact that the default value for @ship_cell was an empty array, instead of nil, so I updated it to nil to match the iteration pattern. Then I had to update line 18 because `.push` is an array method, so it wouldn't execute for the nil value. Instead, I just updated it to `=` to set @ship_cell equal to the object itself.

cell_test rb: 31 - Yeah, this test passes now. The reason the return was originally in brackets was due to the issue I mentioned above. @ship_cell was defaulted to an empty array, so the `.push` method was putting the object inside of that array, instead of setting @ship_cell to equal that object.

lib/ship.rb rb:4. - Yeah, we need to use attr_accessor because the health attribute needs to change when the ship is hit. It was throwing me an error and wouldn't let me change the health attribute if it was set as an attr_reader.
